### DISCUSSION TITLE
lisser la puissance

### DISCUSSION THREAD

**Discussion by toat** [jmcollin78/versatile_thermostat#305] [General]
Salut,

J'ai un comportement que je ne comprends pas bien avec l'utilisation de plusieurs entitées dans Vtherm.

J'ai configué 3 switchs pour mon chauffage, ils finissent pas être synchronisé et s'activent tous au même moment, j'aimerai comprendre pourquoi et savoir si c'est pas un problème de configuration de mon côté.

![vtherm](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/d84f20eb-0667-49da-9863-c8b487fdfae8)

![image](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/3d9a3dae-5e21-49dd-80fb-8415e9a7eef6)

![image](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/c23da1d6-d198-4c29-a4d8-10f50ca88a15)

![image](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/75501080-c593-4c83-ac4a-566788765477)

**Comment by jmcollin78** [comment]
Hello, I see that you havent configured the 2nd heater. This may be the source of the issue. Please fill the heater in the order and try.

**Comment by toat** [comment]
Hi,

Seem ok with the 2nd heated.
Thank you.

**Comment by toat** [comment]
Hi, 

After few days my heaters are enabled in same time :
![image](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/cdcf7d6b-7332-4f86-98f8-f7cf3b5f3de5)
It is yesterday.

**Comment by jmcollin78** [comment]
Can you post me the VTherms attributes ? (Dev Tools / states). This is exactly what we don't want.

**Comment by toat** [comment]
climate.versatile_thermostat_rez_de_chaussee
hvac_modes: heat, off
min_temp: 7
max_temp: 35
target_temp_step: 0.1
preset_modes: none, frost, eco, comfort, boost
current_temperature: 18.7
temperature: 19
hvac_action: heating
preset_mode: eco
is_on: true
hvac_mode: heat
type: null
is_controlled_by_central_mode: true
last_central_mode: null
frost_temp: 5
eco_temp: 19
boost_temp: 23
comfort_temp: 21
frost_away_temp: 7
eco_away_temp: 7
boost_away_temp: 7
comfort_away_temp: 7
power_temp: 13
target_temperature_step: 0.1
ext_current_temperature: -0.9
ac_mode: false
current_power: null
current_power_max: null
saved_preset_mode: eco
saved_target_temp: 19
saved_hvac_mode: null
window_state: off
motion_state: null
overpowering_state: null
presence_state: null
window_auto_state: off
window_bypass_state: false
security_delay_min: 60
security_min_on_percent: 0.5
security_default_on_percent: 0.1
last_temperature_datetime: 2024-01-12T08:46:09.545484+01:00
last_ext_temperature_datetime: 2024-01-12T08:35:11.284700+01:00
security_state: false
minimal_activation_delay_sec: 300
device_power: 1
mean_cycle_power: 0.38
total_energy: 876.4
last_update_datetime: 2024-01-12T09:02:18.956163+01:00
timezone: Europe/Paris
window_sensor_entity_id: null
window_delay_sec: null
window_auto_enabled: false
window_auto_open_threshold: null
window_auto_close_threshold: null
window_auto_max_duration: null
motion_sensor_entity_id: null
presence_sensor_entity_id: null
power_sensor_entity_id: null
max_power_sensor_entity_id: null
temperature_unit: °C
is_device_active: true
ema_temp: 18.72
is_over_switch: true
is_inversed: false
underlying_switch_0: switch.salon_fil_pilote_salon_switch_template
underlying_switch_1: switch.cuisine_fil_pilote_cuisine_switch_template
underlying_switch_2: switch.entree_fil_pilote_entree_switch_template
underlying_switch_3: null
on_percent: 0.38
power_percent: 38
on_time_sec: 682
off_time_sec: 1117
cycle_min: 30
function: tpi
tpi_coef_int: 0.6
tpi_coef_ext: 0.01
friendly_name: Versatile Thermostat rez de chaussée
supported_features: 17

![image](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/a872055c-5fd6-43b6-9f5f-71b5b1facf63)

**Reply by jmcollin78** [reply]
This looks fine. I don't understand the behavior.

**Reply by jmcollin78** [reply]
You just have a very long cycle: `cycle_min: 30`. With electrical heater it is better to have more 5 minutes. 

But this should be related to the behavior constated.

**Comment by toat** [comment]
You meant no more 5 minutes for electrical heater ?

It is an electric underfloor heating, i got 3 zones on the main floor and it take long time to heat. 
I set minimal_activation_delay_sec: 300 and cycle_min: 30 to handle that. is it the right way ?

**Comment by jmcollin78** [comment]
> electric underfloor heating, i got 3 zones on the main floor and it take long time to heat.

ok I have not this point in mind. Maybe you can try 15 min to be a little more reactive. But this will not change the issue.

**Comment by jmcollin78** [comment]
What you can try is to reload the integration and all VTherm config. This is in the Dev Tools / Yaml / Reload configuration / Versatile Thermostat.

We will see if it will resynchronise all VTherm. It should not.

**Comment by toat** [comment]
After reload, it seems ok :
![image](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/9cf16659-7873-4ab3-90eb-0b5997fa09d4)

**Reply by jmcollin78** [reply]
ok that is what is expected. I don't know why they have been synchronized after a while.

If this is reproducing, try to understand in which condition the synchronisation occurs please and send that to me.

Thank you

**Reply by toat** [reply]
Today : 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/26632961/272cfabe-f5a1-4aa1-ab35-290cdeff1c01)

Maybe cause by the fact that minimal_activation_delay_sec is too high regarding to the time range VT decide to heat ?

**Reply by jmcollin78** [reply]
So this is clearly a bug. I don't know why this does this, but this is not normal. If you find what happens or have a clues, i'm interesting. I have the same configuration at home and never experience this. So maybe there is a special condition on your side (maybe minimal_activation_delay_sec is part of the answer).

**Comment by jmcollin78** [comment]
Hello @toat ,

Can you confirm me that the problem is still here and that you use power features of VTherm (shredding) ?

**Comment by toat** [comment]
Hi, 

Problem still there, no power facture used.
